syntax = "proto3";
package terminal;
option optimize_for = LITE_RUNTIME;
option cc_enable_arenas = true;

message TerminalShellRequest {
  string container = 1;
}
message TerminalResizeRequest {
  int64 columns = 1;
  int64 rows = 2;
}

message TerminalInput {
  string data = 1;
}

message TerminalOutput {
  string data = 1;
}

message SessionMessage {
  enum Type {
    ERROR = 0;
    INFO = 1;
    WARNING = 2;
  }

  Type type = 1;
  string data = 2;
}
